--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -51,10 +51,10 @@
 endif()
 
 # execute the command "qmake -query QT_INSTALL_PLUGINS" to get the path of plugins dir.
-execute_process(COMMAND ${QT_QMAKE_EXECUTABLE} -query QT_INSTALL_PLUGINS
-    OUTPUT_VARIABLE QT_PLUGINS_DIR
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
+#execute_process(COMMAND ${QT_QMAKE_EXECUTABLE} -query QT_INSTALL_PLUGINS
+#    OUTPUT_VARIABLE QT_PLUGINS_DIR
+#    OUTPUT_STRIP_TRAILING_WHITESPACE
+#)
 if(QT_PLUGINS_DIR)
     message(STATUS "Qt6 plugin directory:" "${QT_PLUGINS_DIR}")
 else()
